### DISCUSSION TITLE
Snapshots for traces

### DISCUSSION THREAD

**Discussion by bboreham** [grafana/grafana#42285] [Explore]
I would like a feature very like snapshot for dashboards, applied to the trace view.

Storing every trace for a system consumes a lot of storage, so it is typical to expire them after a short period.
However it is also likely that one or two traces can serve as illustration for an incident or optimisation idea, and it would be valuable to preserve them for months.

For illustration, this is the panel I get on Grafana Cloud for a dashboard snapshot:
![image](https://user-images.githubusercontent.com/8125524/143431628-a58f8099-a7f4-4598-b110-7e2a6a85ea66.png)

**Comment by yvrhdn** [comment]
The snapshot feature is designed to share dashboards with external users, so besides caching the results it will also hide underlying datasources.
Do you want the ability to share trace view with external users or are you just interested in caching the results for a long time?

**Reply by bboreham** [reply]
Mostly the long time, but sharing with the world can be useful, say to report an issue on a public GitHub repo.

**Reply by yvrhdn** [reply]
Traces often contain a lot of sensitive information, to simplify sharing traces it would be useful to have a mechanism to filter out tags/attributes from your snapshot.

**Comment by aocenas** [comment]
So I assume what this basically needs is a snapshots for Explore is that right? So grafana would store the data and you could access it later without accessing the data source directly.

**Comment by connorlindsey** [comment]
@bboreham We support downloading traces as JSON ('Inspector' -> 'Data' -> 'Download ...'). Would that work for what you're requesting?

**Comment by bboreham** [comment]
How would I load the data up again to view it?

**Comment by connorlindsey** [comment]
There's a JSON Upload query in Tempo where you can upload the file to view the trace again

**Comment by bboreham** [comment]
This doesn’t sound very convenient for my objective to “serve as illustration for an incident or optimisation idea”. 
I want something I can link to from a Google Doc or GitHub issue.

**Comment by aocenas** [comment]
I think there was also the idea of just having snapshots similar to dashboards but for explore (so would apply also to traces)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #44435. Feel free to continue the discussion around this there. Thank you!
